File: app/Models/Parcel.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Parcel extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var list<string>
     */
    protected $fillable = [
        'user_id',
        'culture_id',
        'nom',
        'surface',
        'date_plantation',
        'date_recolte_estimee',
        'status',
        'health_score',
        'latitude',
        'longitude',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'surface' => 'decimal:4',
            'health_score' => 'float',
            'date_plantation' => 'date',
            'date_recolte_estimee' => 'date',
            'status' => \App\Enums\ParcelStatusEnum::class,
            'latitude' => 'decimal:7',
            'longitude' => 'decimal:7',
        ];
    }

    /**
     * Get the user that owns the parcel.
     */
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Get the culture for this parcel.
     */
    public function culture()
    {
        return $this->belongsTo(Culture::class);
    }

    /**
     * Get the interaction IAs for this parcel.
     */
    public function interactionIas()
    {
        return $this->hasMany(InteractionIA::class);
    }

    /**
     * Get the weather data for this parcel.
     */
    public function weatherData()
    {
        return $this->hasMany(WeatherData::class);
    }

    /**
     * Get the reports generated for this parcel.
     */
    public function reports()
    {
        return $this->hasMany(Report::class);
    }
}

File: mobile/lib/src/models/parcel.dart
import 'package:equatable/equatable.dart';

import 'culture.dart';

/// Mirrors App\Http\Resources\ParcelResource.
class Parcel extends Equatable {
  const Parcel({
    required this.id,
    required this.nom,
    this.surface,
    this.status,
    this.healthScore,
    this.datePlantation,
    this.dateRecolteEstimee,
    this.latitude,
    this.longitude,
    this.createdAt,
    this.culture,
  });

  final int id;
  final String nom;
  final double? surface;
  final String? status;
  final double? healthScore;
  final DateTime? datePlantation;
  final DateTime? dateRecolteEstimee;
  final double? latitude;
  final double? longitude;
  final DateTime? createdAt;
  final Culture? culture;

  factory Parcel.fromJson(Map<String, dynamic> json) {
    return Parcel(
      id: json['id'] as int,
      nom: json['nom'] as String? ?? '',
      surface: (json['surface'] as num?)?.toDouble(),
      status: json['status'] as String?,
      healthScore: (json['health_score'] as num?)?.toDouble(),
      datePlantation: _date(json['date_plantation']),
      dateRecolteEstimee: _date(json['date_recolte_estimee']),
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      createdAt: _date(json['created_at']),
      culture: json['culture'] is Map<String, dynamic>
          ? Culture.fromJson(json['culture'] as Map<String, dynamic>)
          : null,
    );
  }

  static DateTime? _date(dynamic v) =>
      v == null ? null : DateTime.tryParse(v.toString());

  @override
  List<Object?> get props => [id, nom, status, healthScore];
}
